package com.example.ezzeta.ui.theme

import android.app.Activity
import android.os.Build
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.dynamicDarkColorScheme
import androidx.compose.material3.dynamicLightColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext

private val DarkColorScheme = darkColorScheme(
    primary = EzzetaRed,
    onPrimary = Color.White,
    secondary = EzzetaWhite,
    onSecondary = Color.Black,
    tertiary = Color.LightGray,
    onTertiary = Color.Black,
    background = EzzetaBlack,
    onBackground = EzzetaWhite,
    surface = EzzetaDarkGray,
    onSurface = EzzetaWhite,
    surfaceVariant = EzzetaDividerDark,
    onSurfaceVariant = Color.LightGray,
    outline = EzzetaDividerDark
)

private val LightColorScheme = lightColorScheme(
    primary = EzzetaRed,
    onPrimary = Color.White,
    secondary = EzzetaBlack,
    onSecondary = Color.White,
    tertiary = Color.Gray,
    onTertiary = Color.White,
    background = Color.White,
    onBackground = Color.Black,
    surface = EzzetaGray,
    onSurface = Color.Black,
    surfaceVariant = EzzetaDivider,
    onSurfaceVariant = Color.DarkGray,
    outline = EzzetaDivider
)

@Composable
fun EzzetaTheme(
    darkTheme: Boolean = isSystemInDarkTheme(),
    dynamicColor: Boolean = false,
    content: @Composable () -> Unit
) {
    val colorScheme = when {
        dynamicColor && Build.VERSION.SDK_INT >= Build.VERSION_CODES.S -> {
            val context = LocalContext.current
            if (darkTheme) dynamicDarkColorScheme(context) else dynamicLightColorScheme(context)
        }

        darkTheme -> DarkColorScheme
        else -> LightColorScheme
    }

    MaterialTheme(
        colorScheme = colorScheme,
        typography = Typography,
        content = content
    )
}